src/main/java/org/apache/commons/text/cases/CamelCase.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.text.cases;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.apache.commons.lang3.CharUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Case implementation that parses and formats strings of the form 'myCamelCase'
+ * <p>
+ * This case separates tokens on uppercase ascii alpha characters, with the exception
+ * that the first token begin with a lowercase ascii alpha character.
+ * </p>
+ */
+public class CamelCase implements Case {
+
+    /** constant reuseable instance of this case. */
+    public static final CamelCase INSTANCE = new CamelCase();
+
+    /**
+     * Constructs new CamelCase instance.
+     */
+    public CamelCase() {
+        super();
+    }
+
+    /**
+     * Parses string tokens from a Camel Case formatted string.
+     * <p>
+     * Parses each character of the string parameter and creates new tokens when uppercase ascii
+     * letters are encountered. The upppercase letter is considered part of the new token. The very
+     * first character of the string is an exception to this rule and must be a lowercase ascii
```

Comment:
It enforces the rule that the input be a camel cased string (so that `format(parse(<input>)) == <input>` holds true). Not enforcing that rule would be a form of "loose" parsing, as opposed to the current strict logic which expects the user to know the case of the parsed string. Not enforcing this rule essentially makes this a pascal case parser, which already exists, and the API should let the user know that they're not using the correct parser.

If we want to implement separate strict vs loose parsing then I'll give some more thought about how to expose that in a way that makes sense.